src/spikeinterface/extractors/mdaextractors.py

```diff
@@ -202,14 +202,33 @@ def __init__(self, file_path, sampling_frequency):
         sorting_segment = MdaSortingSegment(firings)
         self.add_sorting_segment(sorting_segment)
 
+        # Store the max channel for each unit
+        # Every spike assigned to a unit (label) has the same max channel
+        # ref: https://github.com/SpikeInterface/spikeinterface/issues/3695#issuecomment-2663329006
+        label_to_channel = dict()
+        segment = self._sorting_segments[0]
+        for spike_event_index in range(len(segment._labels)):
+            label = segment._labels[spike_event_index]
+            max_channel = segment._max_channels[spike_event_index]
+            label_to_channel[label] = max_channel
```

Comment:
```suggestion
        max_channels = []
        segment = self._sorting_segments[0]
        for unit_id in self.unit_ids:
            label_mask = segment._labels == unit_id
            # since all max channels are the same, we can just grab the first occurrence for the unit
            max_channel = segment._max_channels[label_mask][0]
            max_channels.append(max_channel)
```
You can just loop over unit ids, to make sure it's already sorted accordingly to the `Sorting.unit_id`.
Also, since max channels are all the same, we can just grab the first value

Reply:
Sounds great